frontend/src/js/components/settings/AddonSelection.tsx

```diff
@@ -0,0 +1,220 @@
+  const currentPlanName = PLANS[currentPlan].name;
+  const { classes } = useStyles();
+  const dispatch = useAppDispatch();
+  const onAddOnClick = (name: string, isAdd: boolean) => {
+    setAction({ name, isAdd });
+  };
+  const requestAddon = async () => {
+    if (!action) return;
+    const { name: addonName, isAdd } = action;
+    let requested_addons = addOnsToString(org.addons).split(', ');
+    if (isAdd) {
+      requested_addons.push(addonName);
+    } else {
+      requested_addons = requested_addons.filter(addon => addon !== addonName);
+    }
+    try {
+      await dispatch(
+        requestPlanChange({
+          tenantId: org.id,
+          content: {
+            current_plan: currentPlanName,
+            requested_plan: currentPlanName,
+            current_addons: addOnsToString(org.addons) || '-',
+            requested_addons: requested_addons.filter(addon => !!addon).join(', ') || '-',
+            user_message: ''
+          }
+        })
+      ).unwrap();
+    } catch (error) {
+      console.error(error);
+      return;
+    }
+    updateLocalStorage(org.id, addonName, isAdd);
+    setAction(null);
+  };
+  const relevantAddons = useMemo(
+    () => {
+      const currentState = JSON.parse(localStorage.getItem(org.id + '_upgrades') || '{}');
+      return Object.entries(ADDONS)
+        .reduce((acc: RelatableAddon[], [addOnName, addOn]) => {
+          const isEnabled = addons.some(orgAddOn => orgAddOn.enabled && addOnName === orgAddOn.name);
+          let pending = currentState[addOnName];
+
+          if (pending && pending.pending && pending.isAdd === isEnabled) {
+            clearLocalStorageEntry(org.id, addOnName);
+            pending = null;
+          }
+          acc.push({ ...addOn, name: addOnName, isEnabled, pending, isEligible: addOn.eligible.indexOf(currentPlan) > -1 });
+          return acc;
+        }, [])
+        .sort((a, b) => a.name.localeCompare(b.name));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [JSON.stringify(addons), JSON.stringify(features), action, org]
+  );
+  return (
+    <>
+      {action && <ConfirmAddon name={action.name} onClose={() => setAction(null)} onConfirm={() => requestAddon()} variant={action.isAdd ? 'add' : 'remove'} />}
+      <h3 className="margin-top-large">Get more features with add-ons</h3>
+
+      <div className="flexbox column">
+        <p>
+          Enhance your Mender plan with optional add-on packages – offering additional features to easily manage your software and devices over-the-air. Below
+          you can request which add-ons you’d like to be included with your plan.
+        </p>
+        <div className={classes.planPanelAddon}>
+          {relevantAddons.map(addOn => {
+            const addonItemClasses = `${addOn.isEnabled ? 'active' : ''} ${addOn.isEligible ? '' : 'muted'} ${classes.planPanelAddonItem}`;
+            return (
+              <>
```

Comment:
```suggestion
              <Fragment key={addOn.name}>
```
to not cause react errors